modules/Core/pages/panel/groups.php

```diff
@@ -294,78 +298,213 @@
 
-                $group = Group::find($_GET['group']);
-                if (!$group) {
-                    Redirect::to(URL::build('/panel/core/groups'));
-                }
+            break;
 
-                if ($group->id == 2 || ((in_array($group->id, $user->getAllGroupIds())) && !$user->hasPermission('admincp.groups.self'))) {
-                    Redirect::to(URL::build('/panel/core/groups'));
-                }
+        case 'clone':
+            if (!isset($_GET['group']) || !is_numeric($_GET['group'])) {
+                Redirect::to(URL::build('/panel/core/groups'));
+            }
 
-                if (Input::exists()) {
-                    $errors = [];
+            $group = Group::find($_GET['group']);
+            if (!$group) {
+                Redirect::to(URL::build('/panel/core/groups'));
+            }
 
-                    if (Token::check()) {
-                        // Token valid
-                        // Build new JSON object for permissions
-                        $perms = [];
-                        if (isset($_POST['permissions']) && count($_POST['permissions'])) {
-                            foreach ($_POST['permissions'] as $permission => $value) {
-                                $perms[$permission] = 1;
-                            }
-                        }
-                        $perms_json = json_encode($perms);
+            if (Input::exists()) {
+                $errors = [];
+                if (Token::check()) {
+                    if (Input::get('action') == 'update') {
+                        $validation = Validate::check($_POST, [
+                            'groupname' => [
+                                Validate::REQUIRED => true,
+                                Validate::MIN => 2,
+                                Validate::MAX => 20
+                            ],
+                            'html' => [
+                                Validate::MAX => 1024
+                            ]
+                        ])->messages([
+                            'groupname' => [
+                                Validate::REQUIRED => $language->get('admin', 'group_name_required'),
+                                Validate::MIN => $language->get('admin', 'group_name_minimum'),
+                                Validate::MAX => $language->get('admin', 'group_name_maximum')
+                            ],
+                            'html' => $language->get('admin', 'html_maximum')
+                        ]);
 
-                        try {
-                            $queries->update('groups', $group->id, ['permissions' => $perms_json]);
+                        if ($validation->passed()) {
+                            try {
+                                if (isset($_POST['default']) && $_POST['default'] == 1) {
+                                    $default = 1;
+                                    $cache->setCache('default_group');
```

Comment:
I assume this was already here but it should be removed since we do it on line 381 anyway